Common/src/linear_algebra/CSysMatrix.cpp

```diff
@@ -202,7 +203,17 @@ void CSysMatrix<ScalarType>::Initialize(unsigned long npoint, unsigned long npoi
     if (row_ptr_prec[iPoint] >= part*nnz_per_part)
       omp_partitions[part++] = iPoint;
   }
-  omp_partitions[omp_num_parts] = nPointDomain;
+
+  for (unsigned long thread = 0; thread < omp_num_parts; ++thread) {
+    const unsigned long begin = omp_partitions[thread];
+    const unsigned long end = omp_partitions[thread + 1];
+    if (begin > end) {
+      SU2_MPI::Error("Invalid nodes are distributed in the thread " + to_string(thread) + ".", CURRENT_FUNCTION);
+    } else if (begin == end) {
+      cout << "WARNING: Redundent thread has been detected. Performance could be impacted." << endl;
+      break;
+    }
```

Comment:
I think it is impossible for begin > end give the way they are computed now.
```suggestion
    const auto begin = omp_partitions[thread];
    const auto end = omp_partitions[thread + 1];
    if (begin == end) {
      cout << "WARNING: Redundant thread has been detected. Performance could be impacted due to low number of nodes per thread." << endl;
      break;
    }
```